### ISSUE TITLE
Incorrect prototype handling

### ISSUE THREAD

**Issue by SeeSpring** [kaleidawave/ezno#207]
```js
let a = {a:0}
let b = {get a(){return 0}}
let c = Math.random() ? a : b
let d = {}
Object.setPrototypeOf(d, c)
let e = ''
if ('a' in d) {
  e = 0
}
e.toUpperCase()
```

Throws `Uncaught TypeError: e.toUpperCase is not a function` at runtime

**Comment by kaleidawave** [comment]
Yes there seems to be some issues with https://kaleidawave.github.io/ezno/playground/?id=klnuy
1. `has_property` could do better on unions or condition results (in the case of the value assigned to `c`). The currently do not look at both sides and figure a result. (the first `print_type("a" in c);` should be `true`)
2. It seems that the `setPrototypeOf` does not register the prototype when the prototype is a conditional type. It works for `setPrototypeOf(d, a)` and `setPrototypeOf(d, b)` but not with `c`. Once  the property access works and returns a result, then `"a" in d` should return `true` (rather than `false` and the example above should work 🤞)

**Comment by kaleidawave** [comment]
Have investigated this just now and I think I have found two problems
1. `setPrototypeOf` is an internal function. Most internal functions shouldn't run if there argument are not constants although some should. Have added it to the list that run if argument is generic (like conditional types are)
2. The access on the prototype looks up "simple object properties", rather than running the general try get property on type. Should change which function is called.

Should have a fix in #209 for it later (should do in separate commit)
